lib/tasks/export_eligible_users_with_outstanding_income_evidences.rake

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'csv'
+# This report generates list of all the applicants with outstanding income evidences between 95 and 160 days
+
+# To generate a csv of users eligible for an income evidence due_on extension without migrating data, run the following:
+# RAILS_ENV=production bundle exec rake reports:export_eligible_users_with_outstanding_income_evidences
+
+# To migrate eligible users' income evidences due_on to their new due date, run the following:
+# RAILS_ENV=production bundle exec rake reports:export_eligible_users_with_outstanding_income_evidences[true]
+
+# The CSV file generated will be called users_with_outstanding_income_evidence_due_dates_between_95_and_160_days.csv, and will be located in the root folder
+
+namespace :reports do
+  task :export_eligible_users_with_outstanding_income_evidences, [:migrate_users] => :environment do |_t, args|
+    file_name = "#{Rails.root}/users_with_outstanding_income_evidence_due_dates_between_95_and_160_days.csv"
+    today = TimeKeeper.date_of_record
+
+    field_names  = [
+      :user_hbx_id,
+      :original_due_date,
+      :due_date_after_extension,
+      :due_date_successfully_extended
+    ]
+
+    applications = FinancialAssistance::Application.where(
+      :"applicants.income_evidence" => { :$exists => true },
+      :"applicants.income_evidence.aasm_state" => { :$eq => "outstanding" },
+      :"applicants.income_evidence.due_on" => { :"$gt" => (today - 160.days), :"$lte" => (today - 95.days) }
+    )
+
+    CSV.open(file_name, "w", write_headers: true, headers: field_names) do |csv|
+      applications.each do |application|
+        application.applicants.each do |applicant|
```

Comment:
@TristanB17 Please add exception handling by logging the errors. We could have some bad data and we do not want to stop the report instead we want to skip the record with bad data.

Reply:
@saikumar9 Can do!

Reply:
@saikumar9 added that exception handling!

Reply:
@saikumar9 added those ids!